packages/specs/schema/src/components/default/discriminatorMappingsMapper.ts

```
@@ -0,0 +1,25 @@
+import type {JsonSchemaOptions} from "../../interfaces/JsonSchemaOptions.js";
+import {registerJsonSchemaMapper} from "../../registries/JsonSchemaMapperContainer.js";
+import {toRef} from "../../utils/ref.js";
+
+function discriminatorMappingsMapper(obj: any, _: JsonSchema, options: JsonSchemaOptions) {
+  if (obj.discriminator) {
+    obj.discriminator.mappings = Object.entries(obj.discriminator.mappings).reduce(
+      (mappings, [key, value]: [string, JsonSchema | string]) => {
+        const ref = isString(value) ? value : toRef(value, null, options).$ref;
+
+        return {
+          ...mappings,
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)